src/stacks-hierarchy/Root_PurchaseConfirmationScreen/components/SelectPaymentMethodSheet.tsx

```diff
@@ -52,59 +56,92 @@ export const SelectPaymentMethodSheet: React.FC<Props> = ({
     currentOptions?.paymentMethod,
   );
 
   return (
-    <BottomSheetContainer
-      title={t(SelectPaymentMethodTexts.header.text)}
-      fullHeight
-    >
+    <BottomSheetContainer title={t(SelectPaymentMethodTexts.header.text)}>
       <View style={{flex: 1}}>
-        <View style={{flexShrink: 1, flexGrow: 1}}>
-          <ScrollView style={styles.paymentMethods}>
-            {defaultPaymentMethods.map((method, index) => {
-              return (
-                <PaymentMethodView
-                  key={method.paymentType}
-                  paymentMethod={method}
-                  shouldSave={shouldSave}
-                  onSetShouldSave={setShouldSave}
-                  selected={
-                    !selectedMethod?.recurringCard &&
-                    selectedMethod?.paymentType === method.paymentType
-                  }
-                  onSelect={(val: PaymentMethod) => {
-                    setSelectedMethod(val);
-                    setShouldSave(false);
-                  }}
-                  index={index}
-                />
-              );
-            })}
-
-            {recurringPaymentMethods && recurringPaymentMethods?.length > 0 && (
-              <View style={styles.listHeading}>
-                <ThemeText>
-                  {t(SelectPaymentMethodTexts.saved_cards.text)}
-                </ThemeText>
-              </View>
-            )}
-            {recurringPaymentMethods?.map((method, index) => (
-              <PaymentMethodView
-                key={method.recurringCard?.id}
+        <View style={styles.paymentMethods}>
+          {singlePaymentMethod.map((method, index) => {
+            return (
+              <SinglePaymentMethod
+                key={method.paymentType}
                 paymentMethod={method}
-                selected={
-                  selectedMethod?.recurringCard?.id === method.recurringCard?.id
-                }
                 shouldSave={shouldSave}
                 onSetShouldSave={setShouldSave}
-                onSelect={setSelectedMethod}
+                selected={
+                  !selectedMethod?.recurringCard &&
+                  selectedMethod?.paymentType === method.paymentType
+                }
+                onSelect={(val: PaymentMethod) => {
+                  setSelectedMethod(val);
+                  setShouldSave(false);
+                }}
                 index={index}
               />
-            ))}
-          </ScrollView>
+            );
+          })}
+          <MultiplePaymentMethodsRadioSection
+            shouldSave={shouldSave}
+            onSetShouldSave={setShouldSave}
+            selected={
+              !selectedMethod?.recurringCard &&
+              selectedMethod?.paymentType ===
+                multiplePaymentMethod[0]?.paymentType
+            }
+            onSelect={() => {
+              setShouldSave(true);
+              setSelectedMethod(multiplePaymentMethod[0]);
```

Comment:
I'm a bit unsure if saving the first payment method here is the best way to do it. Could we use a different state to more clearly communicate what is selected? Something like this maybe:

```ts
type PaymentSelection = {
	kind: 'new_recurring' | 'saved_recurring' | 'non_recurring',
	paymentMethod: PaymentMethod | undefined,
}
```

I haven't looked into the details here, so you might have to change it a bit, but do you think something like it could work? If it instead makes it harder to work with, feel free to ignore 😄